packages/core/src/shipping/strategies/stripe-upe/stripe-upe-shipping-strategy.ts

```diff
@@ -131,28 +139,56 @@ export default class StripeUPEShippingStrategy implements ShippingStrategy {
             appearance,
         });
 
-        const shipping = this._store.getState().shippingAddress.getShippingAddress();
-        const stripeState =
-            shipping?.stateOrProvinceCode && shipping.countryCode
-                ? getStripeState(shipping.countryCode, shipping.stateOrProvinceCode)
-                : shipping?.stateOrProvinceCode;
-        const option = {
+        const shipping = getShippingAddress();
+        const shippingPhoneField = shippingFields.find((field) => field.name === 'phone');
+        let option: StripeElementsCreateOptions = {
+            mode: 'shipping',
             allowedCountries: [availableCountries],
-            defaultValues: {
-                name: shipping?.lastName
-                    ? `${shipping.firstName} ${shipping.lastName}`
-                    : shipping?.firstName || '',
-                address: {
-                    line1: shipping?.address1 || '',
-                    line2: shipping?.address2 || '',
-                    city: shipping?.city || '',
-                    state: stripeState || '',
-                    postal_code: shipping?.postalCode || '',
-                    country: shipping?.countryCode || '',
+            fields: {
+                phone: 'always',
+            },
+            validation: {
+                phone: {
+                    required:
+                        shippingPhoneField && shippingPhoneField.required ? 'always' : 'never',
                 },
             },
         };
 
+        if (shipping) {
+            const {
+                stateOrProvinceCode,
+                countryCode,
+                lastName,
+                firstName,
+                phone,
+                address1,
+                address2,
+                city,
+                postalCode,
+            } = shipping;
+            const stripeState =
+                shipping && stateOrProvinceCode && countryCode
```

Comment:
We have already checked for shipping above? Do we need to check it again?

Reply:
Sorry, I made the change Peng asked for, and I added shipping to remove '?' and then made the small refactor, the error was that I copied the line and forgot to delete my previous change, thank you very much for the observation